zerver/views/streams.py

```diff
@@ -442,6 +443,21 @@ def get_topics_backend(request: HttpRequest, user_profile: UserProfile,
 
     return json_success(dict(topics=result))
 
+@has_request_variables
+def delete_in_topic(request: HttpRequest, user_profile: UserProfile,
+                    stream_id: int=REQ(converter=to_non_negative_int),
+                    topic_name: str=REQ("topic_name")) -> HttpResponse:
+    (stream, recipient, sub) = access_stream_by_id(user_profile, stream_id)
+
+    if user_profile.is_realm_admin:
+        messages = Message.objects.filter(recipient=recipient, subject=topic_name)
+        do_delete_messages(user_profile, messages)
+
+        return json_success()
+
+    else:
+        return json_error(_("This action requires administrative rights"), status=401)
```

Comment:
You should just use `@require_realm_admin` on the view function rather than this if/else block.

As a sidenote, stylistically, it's generally best to write this sort of check with an early-return style, e.g.
```
if not user_profile.is_realm_admin:
    return json_error(_("This action requires administrative rights"), status=401)
￼messages = Message.objects.filter(recipient=recipient, subject=topic_name)
...


```

since then the indentation doesn't go crazy :)